### ISSUE TITLE
Get-DbaRegisteredServer - Cannot find an overload for "RegisteredServersStore" and the argument count: "1".

### ISSUE THREAD

**Issue by maddhatter** [dataplat/dbatools#8785]
### Verified issue does not already exist?

I have searched and found no existing issue

### What error did you receive?

```
writeErrorStream      : True
PSMessageDetails      :
Exception             : System.Exception: Cannot find an overload for "RegisteredServersStore" and the argument count: "1". ---> System.Exception: Cannot find an overload for "RegisteredServersStore" and the argument count: "1". --->
                        System.Management.Automation.MethodException: Cannot find an overload for "RegisteredServersStore" and the argument count: "1".
                           at System.Management.Automation.Adapter.GetBestMethodAndArguments(String methodName, MethodInformation[] methods, PSMethodInvocationConstraints invocationConstraints, Object[] arguments, Object[]& newArguments)
                           at System.Management.Automation.DotNetAdapter.ConstructorInvokeDotNet(Type type, ConstructorInfo[] constructors, Object[] arguments)
                           at Microsoft.PowerShell.Commands.NewObjectCommand.CallConstructor(Type type, ConstructorInfo[] constructors, Object[] args)
                           --- End of inner exception stack trace ---
                           --- End of inner exception stack trace ---
TargetObject          :
CategoryInfo          : InvalidOperation: (:) [Write-Error], Exception
FullyQualifiedErrorId : dbatools_Get-DbaRegServer,Stop-Function
ErrorDetails          : Cannot find an overload for "RegisteredServersStore" and the argument count: "1".
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at Stop-Function, C:\Program Files\WindowsPowerShell\Modules\dbatools\1.1.145\allcommands.ps1: line 97479
                        at Get-DbaRegServer<Process>, C:\Program Files\WindowsPowerShell\Modules\dbatools\1.1.145\allcommands.ps1: line 36582
                        at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {0, 1}
```

### Steps to Reproduce

```powershell
Get-DbaRegisteredServer -SqlInstance ServerName
```


### Please confirm that you are running the most recent version of dbatools
```
1.1.145
```
### Other details or mentions

_No response_

### What PowerShell host was used when producing this error

Windows PowerShell (powershell.exe)

### PowerShell Host Version

```
Name                           Value                                                                                                                                                                                                                                          
----                           -----                                                                                                                                                                                                                                          
PSVersion                      5.1.19041.2364                                                                                                                                                                                                                                 
PSEdition                      Desktop                                                                                                                                                                                                                                        
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                                                                                                        
BuildVersion                   10.0.19041.2364                                                                                                                                                                                                                                
CLRVersion                     4.0.30319.42000                                                                                                                                                                                                                                
WSManStackVersion              3.0                                                                                                                                                                                                                                            
PSRemotingProtocolVersion      2.3                                                                                                                                                                                                                                            
SerializationVersion           1.1.0.1                                                                                                                                                                                                                                        
```


### SQL Server Edition and Build number
```
Microsoft SQL Server 2019 (RTM-CU18-GDR) (KB5021124) - 15.0.4280.7 (X64) 
	Jan 23 2023 12:37:13 
	Copyright (C) 2019 Microsoft Corporation
	Enterprise Edition: Core-based Licensing (64-bit) on Windows Server 2019 Datacenter 10.0 <X64> (Build 17763: ) (Hypervisor)
```

### .NET Framework Version

```

PSChildName                      Version       
-----------                      -------       
v2.0.50727                       2.0.50727.4927
v3.0                             3.0.30729.4926
Windows Communication Foundation 3.0.4506.4926 
Windows Presentation Foundation  3.0.6920.4902 
v3.5                             3.5.30729.4926
Client                           4.8.04084     
Full                             4.8.04084     
Client                           4.0.0.0       
```

**Comment by maddhatter** [comment]
@wsmelton - It works fine with the preview-2.0 release

**Comment by potatoqualitee** [comment]
2.0 is super dope and will fix so many things

**Comment by potatoqualitee** [comment]
i have been closing issues that are fixed by 2.0. if you dont mind, ill close htis one too. i keep opening it then it's already resolved.
